### PR TITLE
DAOS-13976 control: Reset checker reports only for pools requested

### PR DESCRIPTION
**Pull request by kjacque**
If the user starts the checker with --reset, the reset will only apply to reports associated with the pools requested. If the user did not pass in a pool ID, the reset will be applied to all existing reports.

Features: control recovery

Required-githooks: true

### Before requesting gatekeeper:

* [x] Two review approvals and any prior change requests have been resolved.
* [x] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [x] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [x] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [x] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Comment by kjacque** [comment]
Keeping this a draft, as the branch is frozen for now. The failures in CI will actually be fixed whenever we can merge master. I would still appreciate a review on this one.

File: src/control/server/mgmt_check.go [hunk 1]
```diff
@@ -245,9 +248,17 @@ func (svc *mgmtSvc) SystemCheckStart(ctx context.Context, req *mgmtpb.CheckStart
 	}
 
 	if resp.Status > 0 {
-		svc.log.Debug("resetting checker findings DB")
-		if err := svc.sysdb.ResetCheckerData(); err != nil {
-			return nil, errors.Wrap(err, "failed to reset checker finding database")
+		if len(req.Uuids) == 0 {
```
Comment (tanabarr) [inline src/control/server/mgmt_check.go:251]: should the command also accept labels?

File: src/control/system/raft/database_checker.go [hunk 2]
```diff
@@ -74,13 +80,26 @@ func (cdb *CheckerDatabase) updateFinding(finding *checker.Finding) error {
 	return nil
 }
 
+func (cdb *CheckerDatabase) removeFinding(finding *checker.Finding) error {
+	_, found := cdb.Findings[finding.Seq]
```
Comment (tanabarr) [inline src/control/system/raft/database_checker.go:84]: this could be combined with the next line

**Review by tanabarr — APPROVED** [review]

File: src/control/server/mgmt_check.go (hunk 1 above)
Comment (kjacque) [reply src/control/server/mgmt_check.go:251]: The incoming request (via grpc) can include labels. They are translated to UUIDs in `makePoolCheckerCall`. I don't really like the way the Uuids field is overloaded for both of these, but that's a bit outside the scope of this PR.

File: src/control/system/raft/database_checker.go (hunk 2 above)
Comment (kjacque) [reply src/control/system/raft/database_checker.go:84]: Fixed

**Review by Nasf-Fan — APPROVED** [review]
I simply verified the patch, it works as expected.

**Comment by kjacque** [comment]
Failures are all known issues:
- https://daosio.atlassian.net/browse/DAOS-14085
- https://daosio.atlassian.net/browse/DAOS-14072

**Pull request merged**